src/se/terhol/pisemka32/VendorImpl.java

```diff
@@ -0,0 +1,79 @@
+package se.terhol.pisemka32;
+
+import java.io.BufferedWriter;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.util.Collection;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+/**
+ * @author Tereza Holm
+ */
+public class VendorImpl implements Vendor {
+    private SortedMap<Magazine, Double> myMagazines = new TreeMap<>();
+
+    @Override
+    public double getPrice(Magazine magazine) {
+        double price;
+        if (magazine == null || !myMagazines.containsKey(magazine)) {
+            price = -1;
+        } else {
+            price = myMagazines.get(magazine);
+        }
+        return price;
+    }
+
+    @Override
+    public double setPrice(Magazine magazine, double price) {
+        if (price < 0 || magazine == null) {
+            throw new IllegalArgumentException();
+        }
+        double originalPrice;
+        myMagazines.put(magazine, price);
+        if (!myMagazines.containsKey(magazine)) {
+            originalPrice = -1;
+        } else {
+            originalPrice = myMagazines.get(magazine);
+        }
+        return originalPrice;
+    }
+
+    @Override
+    public Collection<Magazine> getMostExpensive() {
+        Collection<Magazine> mostExpensive = new TreeSet<>();
+        double highestPrice = 0;
```

Comment:
You can use a library function for this and delete the for cycle:

```java
double highestPrice = Collections.max(map.values());
```

Reply:
Done